lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
@mscdex Is `args` there an `arguments` object or an `Array`? I'm aware that referencing undefined values on an `arguments` object does have significant overhead, my benchmarks show that that is not the case for a real array.